File: android/src/main/java/com/rebeloid/unity_mediation/StateUtils.java
package com.rebeloid.unity_mediation;

import com.unity3d.mediation.AdState;

public final class StateUtils {

    public static String convertState(AdState state) {
        switch (state) {
            case UNLOADED:
                return "unloaded";
            case LOADING:
                return "loading";
            case LOADED:
                return "loaded";
            case SHOWING:
                return "showing";
            default:
                return "";
        }
    }

    private StateUtils() {
    }
}

File: ios/Classes/StateUtils.swift
import UnityMediationSdk

class StateUtils {
    static func convertState(_ state: UMSAdState) -> String {
        switch (state) {
        case .unloaded:
            return "unloaded";
        case .loading:
            return "loading";
        case .loaded:
            return "loaded";
        case .showing:
            return "showing";
        @unknown default:
            return "";
        }
    }
}
